packages/mdc-segmented-button/segment/_segment.scss

```diff
@@ -0,0 +1,92 @@
+//
+// Copyright 2020 Google Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Core: Define complete styles to render component segment, including base and theme styles
+
+@use "@material/feature-targeting/functions" as feature-targeting-functions;
+@use "@material/feature-targeting/mixins" as feature-targeting-mixins;
+@use './segment-theme';
+
+@mixin core-styles($query: feature-targeting-functions.all()) {
+  // TODO: include test for element resizing besides the left/right border
+  .mdc-segmented-button__segment {
+    @include segment-theme.base($query);
+
+    // Style for when element is selected
+    .mdc-segmented-button__segment--selected {
+      @include segment-theme.selected($query);
+
+      // Avoids double borders
+      + .mdc-segmented-button__segment {
+        @include segment-theme.hide-left-border($query);
+      }
+    }
+
+    // Style for when element is unselected
+    &:not(.mdc-segmented-button__segment--selected) {
```

Comment:
Can we not use `:not()` and use CSS inheritance instead to override unselected styles?